mayastor/src/grpc/mayastor_grpc.rs

```diff
@@ -32,36 +33,75 @@ use crate::{
     nexus_uri::NexusBdevError,
     subsys::PoolConfig,
 };
+use futures::FutureExt;
 use nix::errno::Errno;
 use rpc::mayastor::*;
-use std::{convert::TryFrom, ops::Deref, time::Duration};
+use std::{convert::TryFrom, fmt::Debug, ops::Deref, time::Duration};
 use tonic::{Request, Response, Status};
 #[derive(Debug)]
 struct UnixStream(tokio::net::UnixStream);
 
+use ::function_name::named;
+use std::panic::AssertUnwindSafe;
+
+impl GrpcClientContext {
+    #[track_caller]
+    pub fn new<T>(req: &Request<T>, fid: &str) -> Self
```

Comment:
Im fine with this. However, IMHO, if we have T + Debug, then we don't need the fid. Through the request, we *know* what method it should have been, right? Does it matter? Not really,  but fewer crates to carry around is always better.

Reply:
I am not sure if it's possible to get method's name from tonic::Request, as it primarily represents the method arguments themselves.
In fact, printing the request via {:?} doesn't show any presence of the method name:
```
Request { metadata: MetadataMap { headers: {"te": "trailers", "content-type": "application/grpc", "user-agent": "tonic/0.4.3"} }, message: DestroyPoolRequest { name: "xyz" }, extensions: Extensions }
```

I presume the decision about method name has already been made by a higher-level logic of Tonic before dispatching the request to the method, so the method name is not needed in the request.

Reply:
Yes but, look:
```
DestroyPoolRequest {}...
```

That gives us all the info already, no?

Reply:
Yes but notice that in the Request we see it's a DestroyPoolRequest... so we have all we need?

Reply:
If we don't want to display the name of the method which failed (like: "pool_destroy(): method failed, args: {}", but display only args, then it's fine.

Reply:
As discussed lets leave it as-is :+1: